src/authInfo.ts

```diff
@@ -1096,30 +1101,29 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
     };
   }
 
-  private async retrieveUsername(
-    instanceUrl: string,
-    accessToken: string,
-    orgId: string,
-    userId: string
-  ): Promise<Optional<string>> {
+  private async retrieveUsername(instanceUrl: string, accessToken: string): Promise<Optional<UserInfo>> {
     // Make a REST call for the username directly.  Normally this is done via a connection
     // but we don't want to create circular dependencies or lots of snowflakes
     // within this file to support it.
-    const apiVersion = 'v42.0'; // hardcoding to v42.0 just for this call is okay.
     const instance = ensure(instanceUrl);
-    const url = `${instance}/services/data/${apiVersion}/sobjects/User/${userId}`;
+    const url = `${instance}/services/oauth2/userinfo`;
     const headers = Object.assign({ Authorization: `Bearer ${accessToken}` }, SFDX_HTTP_HEADERS);
-
+    let orgId;
     try {
       this.logger.info(`Sending request for Username after successful auth code exchange to URL: ${url}`);
       const response = await new Transport().httpRequest({ url, headers });
       if (response.statusCode >= 400) {
         this.throwUserGetException(response);
       } else {
-        return asString(parseJsonMap(response.body).Username);
+        const json = parseJsonMap(response.body) as UserInfo;
+        orgId = asString(json.organization_id);
+        return json;
       }
     } catch (err) {
-      throw SfdxError.create('@salesforce/core', 'core', 'AuthCodeUsernameRetrievalError', [orgId, err.message]);
+      throw SfdxError.create('@salesforce/core', 'core', 'AuthCodeUsernameRetrievalError', [
+        orgId || 'unknown',
```

Comment:
You can modify the message to get rid of the orgId. I think it will always be undefined here, because if the API call fails OR the parse response fails, the orgId will never get set.